### PR TITLE
Reorganization of driver-specific code

### PR DESCRIPTION
**Pull request by avelanarius**
#### Overview
This PR fixes some problems regarding Java Driver 3.x-specific code in the library.

Before this change, the library required the users to manually create a Java Driver 3.x session and pass it to the `CDCConsumer`. There could be some problems with driver compatibility, as the library requires quite a recent version of Java Driver 3.x - fixed in this PR by shading Java Driver.

The new `CDCConsumer` API is more easy to use, as you don't have to create a Java Driver 3.x session manually, instead only provide a connection configuration properties.

The organization of the scylla-cdc-java is also improved, by moving Java Driver 3.x-specific code to a separate scylla-cdc-driver3 module.

#### Structure of this PR

This PR can be logically split into 3 phases: (all of them crucial, not divisible into multiple PRs):

1. Prepare the library for shading Java Driver:
1.1. Create a new scylla-cdc-driver3 module: move Java Driver 3.x code to a separate module (without shading yet).
1.2. Change ImmutableList to List: Guava API is now not exposed to the "external world".
1.3. Port UUIDs implementation from Java Driver: UUIDs class is from Java Driver. By porting it to the library, a dependency on Java Driver in scylla-cdc-base is not needed.
1.4. Reorganize Driver3 translators: Translators are classes that take an object returned by Java Driver 3.x and return a scylla-cdc-java library representation (and the reverse way). But because we are shading Java Driver 3.x., those classes will not be useful for external clients (types not compatible between shaded and non-shaded Java Driver). Because the replicator used those classes, a 1:1 copy of one translator is made (other translator could be simply moved).
1.5. Introduce Driver3Session and CQLConfiguration. Introduces Driver3Session class, which constructs a Java Driver 3.x. session from CQLConfiguration without exposing internals of Java Driver 3.x to the external world.
2. Do the shade!
2.1. Shade Java Driver, Netty and Guava
3. Additional refactoring now that shading is complete. Making `CDCConsumer` more user-friendly and adjusting it for new design, which does not expect the user to manually create sessions, but just provide the necessary configuration parameters.
3.1. Rename Master's Connectors to MasterConfiguration: just renaming with better name.
3.2. Rename Worker's Connectors to WorkerConfiguration: just renaming with better name.
3.3. Add a Builder class to MasterConfiguration: making the constructor private and adding a Builder class (with sensible defaults).
3.4. Add a Builder class to WorkerConfiguration: making the constructor private and adding a Builder class (with sensible defaults).
3.5. Reorganize CDCConsumer: Small refactoring of Builder class. CDCConsumer now uses CQLConfiguration class and makes a session (Driver3Session) by itself, instead of requiring the user to provide the session by themselves. See the commit description for full description.

#### Testing

Testing done:
1. Testing Printer example app:
1.1. Basic use
1.2. Running from command line script (`./scylla-cdc-printer` bash script in this repo)
1.3. A scenario with generation change
2. Testing Replicator example app:
2.1. Basic use
2.2. Testing if validation still works (validation if connection is correct, validation if tables exists and have CDC enabled)
2.3. Running from command line script (`./scylla-cdc-replicator` bash script in this repo)
3. Testing Scylla CDC Source Connector:
3.1. On Confluent 6.0.0 with fat JAR
3.2. On Confluent 6.0.0. with Confluent Hub packaging
3.3. On open-source Kafka 2.6.0 with fat JAR
3.4. On open-source Kafka 2.6.0 with Confluent Hub packaging
3.5. Testing if connector properties validation still works (validation if connection is correct, validation if tables exists and have CDC enabled)
4. Unit tests pass (currently there are only Master unit tests, no integration tests in the repo)
5. Testing if library works correctly as a dependency in a newly-created Maven project

### CODE REVIEW

**Comment by avelanarius** [comment]
This PR is not yet rebased onto https://github.com/scylladb/scylla-cdc-java/pull/19 (Add skipping of old generations) - this PR is already big, so better not to clutter it more...

File: scylla-cdc-base/pom.xml [hunk 1]
```diff
@@ -9,4 +9,16 @@
...
+        <guava.version>19.0</guava.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${guava.version}</version>
+        </dependency>
+    </dependencies>
```
Comment (haaawk) [inline scylla-cdc-base/pom.xml:23]: Wouldn't we want to use shaded guava in all modules. What good does it make to shade guava in one module when we use it in another?

Comment (avelanarius) [reply scylla-cdc-base/pom.xml:23]: In scylla-cdc-base and scylla-cdc-lib, Guava is pretty much used only for `Preconditions` (which I assume is stable between Guava releases). The original reasoning behind shading was that the driver-specific code used some Future-related methods from Guava, incompatible with newer Guava versions (therefore in current version of this PR, only Guava in scylla-cdc-driver3 is shaded).

What would be a proper approach to shade Guava in all these three modules (scylla-cdc-base, scylla-cdc-lib, scylla-cdc-driver3)? 

- Shade it in top-level one (scylla-cdc-base) and then in other modules carefully import shaded classes from scylla-cdc-base (no shading in those other modules)?
- Do the shading separately in each module?

Comment (haaawk) [reply scylla-cdc-base/pom.xml:23]: Yes. Shading it in a module used by everything else would be best.

**Comment by avelanarius** [comment]
(force-pushed rebased onto latest master)

**Review by haaawk — APPROVED** [review]

**Pull request merged**